src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,125 @@
+ * @param seen1ESPTOrganizations 
+ * @returns 
+ */
+export async function getCached1ESPTSchema(context: vscode.ExtensionContext, organizationName: string, session: AzureSession, lastUpdated1ESPTSchema: Map<string, Date>, seen1ESPTOrganizations: Set<string>): Promise<URI> {
+    if (seen1ESPTOrganizations.has(organizationName)) {
+        const schemaUri1ESPT = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+
+        try {
+            if (session.userId.endsWith("@microsoft.com")) {
+                if ((new Date().getTime() - lastUpdated1ESPTSchema.get(organizationName).getTime()) < milliseconds24hours) {
+                    const schemaFileExists = await vscode.workspace.fs.stat(schemaUri1ESPT);
+                    if (schemaFileExists) {
+                        logger.log("Returning cached schema for 1ESPT", 'SchemaDetection');
+                        return schemaUri1ESPT;
+                    }
+                }
+                // schema is older than 24 hours, fetch schema file again
+                else {
+                    logger.log(`Skipping cached 1ESPT schema for ${organizationName} as it is older than 24 hours`, `SchemaDetection`);
+                }
+            }
+            else {
+                const signInAction = await vscode.window.showInformationMessage(Messages.notUsing1ESPTSchemaAsUserNotSignedInMessage, Messages.signInLabel);
+                if (signInAction == Messages.signInLabel) {
+                    await vscode.window.withProgress({
+                        location: vscode.ProgressLocation.Notification,
+                        title: Messages.waitForAzureSignIn,
+                    }, async () => {
+                        await vscode.commands.executeCommand("azure-account.login");
+                    });
+                }
+                logger.log(`Skipping cached 1ESPT schema for ${organizationName} as user is not signed in with Microsoft account`, `SchemaDetection`);
+            }
+        }
+        catch (error) {
+            logger.log(`Error : ${error} while fetching cached 1ESPT schema for org: ${organizationName}. It's possible that the schema does not exist.`, 'SchemaDetection');
+        }
+    }
+    return undefined;
+}
+
+/**
+ * User is eligible for 1ESPT schema if 1ESPT schema is available in ADO organization
+ * @param azureDevOpsClient 
+ * @param organizationName 
+ * @returns 
+ */
+export async function get1ESPTRepoIdIfAvailable(azureDevOpsClient: azdev.WebApi, organizationName: string): Promise<string> {
+    try {
+        const gitApi = await azureDevOpsClient.getGitApi();
+        const repositories = await gitApi.getRepositories('1ESPipelineTemplates');
+        if (!repositories || repositories.length === 0) {
+            logger.log(`1ESPipelineTemplates ADO project not found for org ${organizationName}`, `SchemaDetection`);
+            return ""; // 1ESPT ADO project not found
```

Comment:
Make all of these ""s return `undefined` instead

Reply:
Cannot do undefined as we dont want to run the 'get1ESPTRepoIdIfAvailable' multiple times. We initialize repoId1espt as undefined, then check if it's available. If it's available, return the repo id else return empty string. We use the length of repoId to confirm if 1espt repo is available or not. a value of 'not undefined' tells us we have already evaluated